### PR TITLE
Add stylelint to repository

### PR DESCRIPTION
**Pull request by belcherj**

### CODE REVIEW

File: .stylelintrc [hunk 1]
```diff
@@ -0,0 +1,59 @@
+{
+  "rules": {
+    "color-hex-case": "lower",
```
Comment (belcherj) [inline .stylelintrc]: Colors should user lowercase: #ffffff not #FFFFFF

File: .stylelintrc [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    "color-no-invalid-hex": true,
```
Comment (belcherj) [inline .stylelintrc]: Cannot use invalid colors

File: .stylelintrc [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    "function-calc-no-unspaced-operator": true,
```
Comment (belcherj) [inline .stylelintrc]: This: a { top: calc(1px + 2px); } Not: a { top: calc(1px+2px); }

File: .stylelintrc [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    "function-comma-space-after": "always-single-line",
```
Comment (belcherj) [inline .stylelintrc]: This rgba(1, 1, 1, 1) Not: rgba(1,1,1,1)

File: .stylelintrc [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    "function-comma-space-before": "never",
```
Comment (belcherj) [inline .stylelintrc]: This: a { transform: translate(1 , 1) } Not: a { transform: translate(1, 1) }

File: .stylelintrc [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    "function-name-case": "lower",
```
Comment (belcherj) [inline .stylelintrc]: This: a { width: calc(5% - 10em); } Not: a { width: Calc(5% - 10em); }

File: .stylelintrc [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    "function-url-quotes": "always",
```
Comment (belcherj) [inline .stylelintrc]: This: a { background: url('x.jpg') } not a { background: url(x.jpg) }

File: .stylelintrc [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    "function-whitespace-after": "always",
```
Comment (belcherj) [inline .stylelintrc]: This: a { transform: translate(1, 1) scale(3); } Not: a { transform: translate(1, 1)scale(3); }

File: .stylelintrc [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+    "number-leading-zero": "always",
```
Comment (belcherj) [inline .stylelintrc]: This: a { line-height: 0.5; } Not: a { line-height: .5; }

File: .stylelintrc [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+    "number-no-trailing-zeros": true,
```
Comment (belcherj) [inline .stylelintrc]: This: a { line-height: 0.5; } Not: a { line-height: 0.50; }

File: .stylelintrc [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+    "length-zero-no-unit": true,
```
Comment (belcherj) [inline .stylelintrc]: This: a { top: 0; } Not: a { top: 0px; }

File: .stylelintrc [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+
+    "string-no-newline": true,
```
Comment (belcherj) [inline .stylelintrc]: Can't do this:
a {
  content: "first
    second";
}

File: .stylelintrc [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+    "string-quotes": "single",
```
Comment (belcherj) [inline .stylelintrc]: Use single quotes: '' Not: ""

File: .stylelintrc [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+
+    "unit-case": "lower",
```
Comment (belcherj) [inline .stylelintrc]: This: 0px Not: 0PX

**Comment by belcherj** [comment]
Stylelint checks your Sass against the code style guide.  I have included the config from Calypso but can make changes.  I have commented on all of the rules so that you can review.

There are many errors in the Sass using this config.  You may consider making some warnings.  Once the Stylelint is in I am willing to help with the cleanup needed to get to zero errors.

File: .stylelintrc [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+    "unit-no-unknown": true,
```
Comment (belcherj) [inline .stylelintrc]: Cannot use units that are not defined

File: .stylelintrc [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+    "unit-whitelist": ["px", "%", "deg", "ms", "em", "vh", "vw", "rem"],
+
+    "value-list-comma-space-after": "always-single-line",
```
Comment (belcherj) [inline .stylelintrc]: This: a { background-size: 0, 0; } Not: a { background-size: 0,0; }

File: .stylelintrc [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+    "value-list-comma-space-before": "never",
```
Comment (belcherj) [inline .stylelintrc]: This: a { background-size: 0, 0; } Not: a { background-size: 0 , 0; }

File: .stylelintrc [hunk 18]
```diff
@@ -0,0 +1,59 @@
...
+
+    "shorthand-property-no-redundant-values": true,
```
Comment (belcherj) [inline .stylelintrc]: This: .foo { margin: 1px; } Not: .foo { margin: 1px 1px 1px 1px; }

File: .stylelintrc [hunk 19]
```diff
@@ -0,0 +1,59 @@
...
+
+    "property-case": "lower",
```
Comment (belcherj) [inline .stylelintrc]: This: a { width: 1px; } Not: a { Width: 1px; }

File: .stylelintrc [hunk 20]
```diff
@@ -0,0 +1,59 @@
...
+
+    "declaration-block-no-duplicate-properties": true,
```
Comment (belcherj) [inline .stylelintrc]: Cannot define the same property twice in a selector.  Cannot define background twice.

File: .stylelintrc [hunk 21]
```diff
@@ -0,0 +1,59 @@
...
+    "declaration-block-no-ignored-properties": true,
```
Comment (belcherj) [inline .stylelintrc]: http://stylelint.io/user-guide/rules/declaration-block-no-ignored-properties/

File: .stylelintrc [hunk 22]
```diff
@@ -0,0 +1,59 @@
...
+    "declaration-block-trailing-semicolon": "always",
```
Comment (belcherj) [inline .stylelintrc]: This margin: 0; Not: margin: 0

File: .stylelintrc [hunk 23]
```diff
@@ -0,0 +1,59 @@
...
+    "declaration-block-single-line-max-declarations": 0,
```
Comment (belcherj) [inline .stylelintrc]: Don't do this: margin: 0; padding: 0;
Each property must be on a new line.

File: .stylelintrc [hunk 24]
```diff
@@ -0,0 +1,59 @@
...
+    "declaration-block-semicolon-space-before": "never",
```
Comment (belcherj) [inline .stylelintrc]: This: a { color: pink; } Not: a { color: pink ; }

File: .stylelintrc [hunk 25]
```diff
@@ -0,0 +1,59 @@
...
+    "declaration-block-semicolon-space-after": "always-single-line",
```
Comment (belcherj) [inline .stylelintrc]: Space after a semi-colon if anything after it.  Sorta obsolete due to: declaration-block-single-line-max-declarations

File: .stylelintrc [hunk 26]
```diff
@@ -0,0 +1,59 @@
...
+    "declaration-block-semicolon-newline-before": "never-multi-line",
```
Comment (belcherj) [inline .stylelintrc]: Don't do this:
  a {
    color: pink
    ; top: 0;
  }

File: .stylelintrc [hunk 27]
```diff
@@ -0,0 +1,59 @@
...
+    "declaration-block-semicolon-newline-after": "always-multi-line",
```
Comment (belcherj) [inline .stylelintrc]: Must be a newline after semi-colon

File: .stylelintrc [hunk 28]
```diff
@@ -0,0 +1,59 @@
...
+
+    "block-closing-brace-newline-after": "always",
```
Comment (belcherj) [inline .stylelintrc]: Must be a new line after a closing brace

File: .stylelintrc [hunk 29]
```diff
@@ -0,0 +1,59 @@
...
+    "block-closing-brace-newline-before": "always-multi-line",
```
Comment (belcherj) [inline .stylelintrc]: Must be a newline before a closing brace

File: .stylelintrc [hunk 30]
```diff
@@ -0,0 +1,59 @@
...
+    "block-no-empty": true,
```
Comment (belcherj) [inline .stylelintrc]: No empty blocks: a {}

File: .stylelintrc [hunk 31]
```diff
@@ -0,0 +1,59 @@
...
+    "block-opening-brace-newline-after": "always-multi-line",
```
Comment (belcherj) [inline .stylelintrc]: Must be a new line after an opening brace

File: .stylelintrc [hunk 32]
```diff
@@ -0,0 +1,59 @@
...
+    "block-opening-brace-space-before": "always",
```
Comment (belcherj) [inline .stylelintrc]: Must be a space before and opening brace:  a { Not: a{

File: .stylelintrc [hunk 33]
```diff
@@ -0,0 +1,59 @@
...
+
+    "selector-attribute-brackets-space-inside": "never",
```
Comment (belcherj) [inline .stylelintrc]: This: .foo[target=_blank] Not: .foo[ target=_blank ]

File: .stylelintrc [hunk 34]
```diff
@@ -0,0 +1,59 @@
...
+    "selector-attribute-operator-space-after": "never",
```
Comment (belcherj) [inline .stylelintrc]: This: .foo[target=_blank] Not: .foo[target= _blank]

File: .stylelintrc [hunk 35]
```diff
@@ -0,0 +1,59 @@
...
+    "selector-attribute-operator-space-before": "never",
```
Comment (belcherj) [inline .stylelintrc]: This: .foo[target=_blank] Not: .foo[target =_blank]

File: .stylelintrc [hunk 36]
```diff
@@ -0,0 +1,59 @@
...
+    "selector-combinator-space-after": "always",
```
Comment (belcherj) [inline .stylelintrc]: http://stylelint.io/user-guide/rules/selector-combinator-space-after/

File: .stylelintrc [hunk 37]
```diff
@@ -0,0 +1,59 @@
...
+    "selector-combinator-space-before": "always",
```
Comment (belcherj) [inline .stylelintrc]: http://stylelint.io/user-guide/rules/selector-combinator-space-before/

File: .stylelintrc [hunk 38]
```diff
@@ -0,0 +1,59 @@
...
+    "selector-pseudo-class-case": "lower",
```
Comment (belcherj) [inline .stylelintrc]: This: a:hover {} Not: a:Hover {}

File: .stylelintrc [hunk 39]
```diff
@@ -0,0 +1,59 @@
...
+    "selector-pseudo-class-parentheses-space-inside": "always",
```
Comment (belcherj) [inline .stylelintrc]: This: input:not( [type="submit"] ) {} Not: input:not([type="submit"]) {}

File: .stylelintrc [hunk 40]
```diff
@@ -0,0 +1,59 @@
...
+    "selector-pseudo-element-case": "lower",
```
Comment (belcherj) [inline .stylelintrc]: This: a::before {} Not: a::Before {}

File: .stylelintrc [hunk 41]
```diff
@@ -0,0 +1,59 @@
...
+    "selector-pseudo-element-colon-notation": "double",
```
Comment (belcherj) [inline .stylelintrc]: This: a::before {} Not: a:before {}

File: .stylelintrc [hunk 42]
```diff
@@ -0,0 +1,59 @@
...
+    "selector-pseudo-element-no-unknown": true,
```
Comment (belcherj) [inline .stylelintrc]: No unknown pseudo selectors

File: .stylelintrc [hunk 43]
```diff
@@ -0,0 +1,59 @@
...
+    "selector-type-case": "lower",
```
Comment (belcherj) [inline .stylelintrc]: This: a {} Not: A {}

File: .stylelintrc [hunk 44]
```diff
@@ -0,0 +1,59 @@
...
+    "selector-no-id": true
```
Comment (belcherj) [inline .stylelintrc]: Don't style on ID's, may need to make this a warning and fix over time.  Let me know if you want an explanation of why you shouldn't style on id's

**Comment by mikejolley** [comment]
@jameskoster

**Comment by jameskoster** [comment]
Looks good to me.

**Pull request merged**